### ISSUE TITLE
Possible code smell

### ISSUE THREAD

**Issue by bennyg123** [pauldangpaul/WigBad#2]
https://github.com/pauldangpaul/WigBad/blob/709a5137a29a2f43c10a2fd7979e51f3adf14868/src/components/App.js#L32

This method seems like a code smell since the state value does not update immediately. For the purpose of what you are trying to do it's better to change it to a regular variable.

**Comment by pauldangpaul** [comment]
So the supposed rule of thumb is as follows?

> _"Use state for any changes that will require a re-render - Cyndaquil "_

**Comment by pauldangpaul** [comment]
Makes sense. And should I need to pass a function to a lower component, I would write a function from the top-level component which modifies the local / class variable.

e.g.

Top level component | 
- has local variable **foodIdCounter**
- has local function **getFoodIdCounter( )**
- has local function **incrementFoodIdCounter( )**
- has local function **clearAllFoodIds( )**

all functions can be passed down as props

**Comment by bennyg123** [comment]
> Makes sense. And should I need to pass a function to a lower component, I would write a function from the top-level component which modifies the local / class variable.
> 
> e.g.
> 
> Top level component |
> 
> * has local variable **foodIdCounter**
> * has local function **getFoodIdCounter( )**
> * has local function **incrementFoodIdCounter( )**
> * has local function **clearAllFoodIds( )**
> 
> all functions can be passed down as props

Yep
